### ISSUE TITLE
Image not rendering on page while the amp-img is out of the viewport

### ISSUE THREAD

**Issue by farshidbeheshti** [ampproject/amphtml#21926]
While the `<amp-img>`  in `<amp-carousel>`  is out of the viewport, it won't be rendered. Inspecting with the devtools,  noticed the img tag is not inserted between the opening and closing `amp-img` tag while the images are out of the viewport, so the images won't be rendered on the page. What is workaround?
See the attachment!
![screen-shot](https://user-images.githubusercontent.com/1665275/56473156-4e117500-647c-11e9-8723-b5447b8ebb27.PNG)

You may want to look at that, here:
http://test.bia2aroosi.com/amp/karaj

**Comment by techhtml** [comment]
Hey, How about adding a `display:block` to your a tag?
I think your `amp-img` don't have enough space, So they forgive rendering

**Comment by farshidbeheshti** [comment]
>How about adding a `display:block` to your a tag?
I don't think so, since of the css class which `amp-carousel` applied to the slides. The slides all have `display:inline-blobk` (please find the attachment again!). 
![image](https://user-images.githubusercontent.com/1665275/56501587-971d0400-6524-11e9-902c-e83bdaedb93b.png)

> I think your `amp-img` don't have enough space, So they forgive rendering
Notice the slides inside the viewport are rendered (with the same markup).

Just as a note: type of the `amp-carousel`  I used is 'carousel', but it works well in `type='slides'` and all images are rendered on a lazy-load way when they scroll into the viewport (again with the same markup)

Farshid

**Comment by jpettitt** [comment]
This is a bug in `rtl` handling of lazy loading. If I remove the `direction: rtl` from the body it works.

**Comment by cathyxz** [comment]
Our `rtl` support has been known to be slightly problematic in `amp-carousel` per https://github.com/ampproject/amphtml/issues/16605, https://github.com/ampproject/amphtml/issues/5950. I think / hope this is being addressed in the carousel v2, can @sparhami or @nainar confirm?

**Comment by aghassemi** [comment]
I think this is worth fixing in current Carousel @sparhami. Likely is it only an issue with `type=carousel` and not `type=slide`

**Comment by lvberkum** [comment]
I would like to second that, it is important to fix this in the current carousel. As it stands amp-carousel is not very useful, as images outside the initial viewport load very erratically.

**Comment by sparhami** [comment]
> I would like to second that, it is important to fix this in the current carousel. As it stands amp-carousel is not very useful, as images outside the initial viewport load very erratically.

This issue should be fixed in amp-carousel 0.2, on Chrome 77+, Safari and Firefox. Let me know if you still see this issue with amp-carousel 0.2, with a page and steps for reproducing the issue.

**Comment by lvberkum** [comment]
I think this problem maybe only occurs with a large number of lengthy carousels, where a significant proportion of the images is outside of the initial viewport?

https://www.amsterdamhotspots.com/index.html (amp-carousel-0.1)

and

https://www.amsterdamhotspots.com/index2.html  (amp-carousel-0.2)

show the same problems:

- 2 carousels outside of the initial viewport will not render, at least not until an arbitrary element is clicked

- the second to last image of each carousel (adsense ad), when outside of the viewport, will not render either. Multiple ads within the same carousel within the viewport render immediately.

**Comment by sparhami** [comment]
For carousel 0.1, there is an experiment flag to make this work in the current dev channel, though I don't know when this will go to production. Going to https://cdn.ampproject.org/experiments.html and enabling the dev channel, the amp-carousel 0.1 version is fixed.

For carousel 0.2, the failure to load is caused by #25568. This should be fixed in the next release. Going to https://cdn.ampproject.org/experiments.html and enabling the dev channel, the amp-carousel 0.2 version is fixed.

**Comment by lvberkum** [comment]
Is there a way to find out when either version will go to production? I am asking because this implementation of amp-carousel runs in a production environment, fixing this is time critical.

**Comment by sparhami** [comment]
> Is there a way to find out when either version will go to production? I am asking because this implementation of amp-carousel runs in a production environment, fixing this is time critical.

Generally, builds go to production on each Tuesday. Some weeks may not have releases, I believe next week will not have a release.

You can join the [AMP Slack chat](amphtml.slack.com) #release channel, which is updated with the latest release status.

**Comment by cathyxz** [comment]
Also this: https://github.com/ampproject/amphtml/blob/master/contributing/release-schedule.md#determining-if-your-change-is-in-production. =)

**Comment by lvberkum** [comment]
Very useful, thanks.
